detection_rules/rule_validators.py

```diff
@@ -445,89 +446,189 @@ def validate_stack_combos(self, data: QueryRuleData, meta: RuleMeta) -> EQL_ERRO
-            err_trailer = (
-                f"stack: {stack_version}, integration: {integration},"
-                f"ecs: {ecs_version}, package: {package}, package_version: {package_version}"
+        # Function to extract the field name from an error message
+        def _subquery_trailer_builder(pkg: str, integ: str | None, pkg_ver: str, stk_ver: str, ecs_ver: str) -> str:
+            return (
+                f"Subquery schema mismatch. "
+                f"package: {pkg}, integration: {integ}, package_version: {pkg_ver}, "
+                f"stack: {stk_ver}, ecs: {ecs_ver}"
             )
 
-            # Validate the query against the schema
-            exc = self.validate_query_with_schema(
-                data=data,
-                schema=eql_schema,
-                err_trailer=err_trailer,
-                min_stack_version=meta.min_stack_version,  # type: ignore[reportArgumentType]
+        # Non-sequence: validate full query against each integration schema
+        def _full_query_trailer_builder(pkg: str, integ: str | None, pkg_ver: str, stk_ver: str, ecs_ver: str) -> str:
+            return (
+                f"Try adding event.module or event.dataset to specify integration module\n\t"
+                f"Will check against integrations {meta.integration} combined.\n\t"
+                f"package: {pkg}, integration: {integ}, package_version: {pkg_ver}, stack: {stk_ver}, ecs: {ecs_ver}"
             )
 
-            if isinstance(exc, eql.EqlParseError):
-                message = exc.error_msg  # type: ignore[reportUnknownVariableType]
-                if message == "Unknown field" or "Field not recognized" in message:
-                    field = extract_error_field(self.query, exc)
-                    trailer = (
-                        f"\n\tTry adding event.module or data_stream.dataset to specify integration module\n\t"
-                        f"Will check against integrations {meta.integration} combined.\n\t"
-                        f"{package=}, {integration=}, {package_version=}, "
-                        f"{stack_version=}, {ecs_version=}"
-                    )
-                    error_fields[field] = {
-                        "error": exc,
-                        "trailer": trailer,
-                        "package": package,
-                        "integration": integration,
-                    }
-                    if data.get("notify", False):
-                        print(f"\nWarning: `{field}` in `{data.name}` not found in schema. {trailer}")
-                else:
+        # Function to build a minimal synthetic sequence containing the subquery
+        def _build_synthetic_sequence_from_subquery(subquery: "ast.SubqueryBy") -> str:
+            subquery_text = str(subquery)
+            join_fields = [str(j) for j in (getattr(subquery, "join_values", []) or [])]
+            dummy_by = f" by {', '.join(join_fields)}" if join_fields else ""
```

Comment:
```suggestion
            join_fields = ", ".join(map(str, getattr(subquery, "join_values", []) or []))
            dummy_by = f" by {join_fields}" if join_fields else ""
```
Nit for readability, might be better (not attached to it either way)

Reply:
<img width="1549" height="603" alt="image" src="https://github.com/user-attachments/assets/6c39d727-edf7-4a65-86ac-088242e0862c" />